cw-orch/src/interchain/interchain_channel_builder.rs

```diff
@@ -0,0 +1,233 @@
+
+use ibc_chain_registry::chain::ChainData;
+use crate::networks::parse_network;
+use crate::InterchainInfrastructure;
+use crate::follow_ibc_execution::follow_trail;
+use crate::daemon::queriers::DaemonQuerier;
+use ibc_chain_registry::chain::Grpc;
+use ibc_relayer_types::core::ics24_host::identifier::ChainId;
+use tonic::transport::Channel;
+use crate::daemon::channel::GrpcChannel;
+use crate::interchain::IcResult;
+use crate::interchain::docker::DockerHelper;
+use crate::interchain::interchain_channel::InterchainChannel;
+use crate::queriers::Ibc;
+use crate::{DaemonError, Hermes};
+
+use super::interchain_channel::InterchainPort;
+
+
+#[derive(Default)]
+struct ChainChannelBuilder{
```

Comment:
This struct allows one to create an Interchain Channel object from different parameter types